src/Types/BigIntType.php

```diff
@@ -47,15 +48,24 @@ public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int
             return $value;
         }
 
-        if ($value > PHP_INT_MIN && $value < PHP_INT_MAX) {
-            return (int) $value;
-        }
-
         assert(
             is_string($value),
             'DBAL assumes values outside of the integer range to be returned as string by the database driver.',
         );
 
+        $dotPos = strpos($value, '.');
```

Comment:
Why do we expect a dot in an integer value coming from the database?

Reply:
see https://dbfiddle.uk/6OSky-ka - if database column type (ex. `decimal`) does not match the used type (ex. `bigint`) the value must be parsed correctly